specs/TryUpdateRuntime.md

```diff
@@ -0,0 +1,229 @@
+    CreateCoreWebView2EnvironmentWithOptions(nullptr, nullptr, options.Get(),
+        Callback<ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler>(
+            [callback, minimalVersionRequired](HRESULT result, ICoreWebView2Environment* environment) -> HRESULT {
+                wil::com_ptr<ICoreWebView2Environment> webViewEnvironment = environment;
+                auto experimentalEnvironment3 =
+                webViewEnvironment.try_query<ICoreWebView2ExperimentalEnvironment3>();
+                HRESULT hr = experimentalEnvironment3->TryUpdateRuntime(
+                    Callback<ICoreWebView2ExperimentalTryUpdateRuntimeCompletedHandler>(
+                        [callback, minimalVersionRequired, experimentalEnvironment3](HRESULT errorCode,
+                           ICoreWebView2ExperimentalUpdateRuntimeResult* result) -> HRESULT {
+                            COREWEBVIEW2_RUNTIME_UPDATE_STATUS updateStatus =
+                                COREWEBVIEW2_RUNTIME_UPDATE_STATUS_FAILED;
+                            if ((errorCode == S_OK) && result)
+                            {
+                              CHECK_FAILURE(result->get_UpdateRuntimeStatus(&updateStatus));
+                            }
+                            if (updateStatus != COREWEBVIEW2_RUNTIME_UPDATE_STATUS_UPDATED)
+                            {
+                              callback(false);
+                            }
+                            else
+                            {
+                              callback(IsCurrentVersionSameOrNewer(minimalVersionRequired));
+                            }
+                            return S_OK;
+                        })
+                        .Get());
+                return S_OK;
+            })
+            .Get());
+}
+
+```
+
+# Remarks
+See comments in [API Details](#api-details) section below.
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+```IDL
+/// Status of TryUpdateRuntime operation result.
+[v1_enum] typedef enum COREWEBVIEW2_UPDATE_RUNTIME_STATUS {
+
+  /// No update for Edge WebView2 Runtime is available.
+  /// Latest version of Edge WebView2 Runtime is already installed.
+  COREWEBVIEW2_UPDATE_RUNTIME_STATUS_NO_UPDATE,
+
+  /// Edge WebView2 Runtime is updated successfully.
+  COREWEBVIEW2_UPDATE_RUNTIME_STATUS_UPDATED,
+
+  /// Edge WebView2 Runtime update is blocked by group policy.
+  COREWEBVIEW2_UPDATE_RUNTIME_STATUS_BLOCKED_BY_POLICY,
+
+  /// Edge WebView2 Runtime update failed.
+  /// See `UpdateError` property of UpdateRuntimeResult for more
+  /// information about the failure.
+  COREWEBVIEW2_UPDATE_RUNTIME_STATUS_FAILED,
+} COREWEBVIEW2_UPDATE_RUNTIME_STATUS;
+
+
+/// The TryUpdateRuntime operation result.
+[uuid(DD503E49-AB19-47C0-B2AD-6DDD09CC3E3A), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalUpdateRuntimeResult : IUnknown {
```

Comment:
Will "Experimental" go away when this exits experimental phase?

Reply:
That is correct.